src/lib/uptime_service/uptime_service.ml

```diff
@@ -63,44 +61,58 @@ let send_uptime_data ~logger ~(submitter_keypair : Keypair.t) ~url ~state_hash
   in
   let headers = Cohttp.Header.of_list [("Content-Type", "application/json")] in
   match%map
-    Monitor.try_with ~here:[%here] ~extract_exn:true (fun () ->
-        match%map
-          Cohttp_async.Client.post ~headers
-            ~body:(Yojson.Safe.to_string json |> Cohttp_async.Body.of_string)
-            url
-        with
-        | {status; _}, body ->
-            if Cohttp.Code.code_of_status status = 200 then
-              [%log info]
-                "Sent block with state hash $state_hash to uptime service at \
-                 URL $url"
-                ~metadata:
-                  [ ("state_hash", State_hash.to_yojson state_hash)
-                  ; ( "includes_snark_work"
-                    , `Bool (Option.is_some block_data.snark_work) )
-                  ; ("url", `String (Uri.to_string url)) ]
-            else
-              let base_metadata =
-                [ ("state_hash", State_hash.to_yojson state_hash)
-                ; ("url", `String (Uri.to_string url))
-                ; ("http_code", `Int (Cohttp.Code.code_of_status status))
-                ; ("http_error", `String (Cohttp.Code.string_of_status status))
-                ; ("payload", json) ]
-              in
-              let extra_metadata =
-                match body with
-                | `String s ->
-                    [("error", `String s)]
-                | `Strings ss ->
-                    [("error", `List (List.map ss ~f:(fun s -> `String s)))]
-                | `Empty | `Pipe _ ->
-                    []
-              in
-              let metadata = base_metadata @ extra_metadata in
-              [%log error]
-                "Failure when sending block with state hash $state_hash to \
-                 uptime service at URL $url"
-                ~metadata )
+    make_interruptible
```

Comment:
Can you uplift this and make the whole checking loop interruptible? For example, if we're still waiting on snark work but we have entered the next 5-slot interval, it would be better to stop there instead